### DISCUSSION TITLE
FluentStorageProvider unpublishes workflows

### DISCUSSION THREAD

**Discussion by Serg046** [elsa-workflows/elsa-core#5531] [Q&A]
I created a simple workflow that listents to http requests and then answers. Then I load it from json via `UseFluentStorageProvider()`. The designer shows the workflow is loaded, then I publish. Workflow is now running and listenning to requests, everything works. But if I stop and run the app again. The workflow is unpublished again and my previous settings are reset. How should I deal with this? I expect that published/unpublished state is managed by an admin bia the designer.

**Comment by glime-ai** [comment]
To address the issue where your workflow is unpublished after restarting the application, you should ensure that the workflow's published state is being persisted correctly. When you publish a workflow from the designer, the state should be saved to the underlying persistence store configured for Elsa. If the published state is not being retained after an application restart, it could be due to a misconfiguration or an issue with the persistence provider.

Here are some steps you can take to troubleshoot and resolve the issue:

1. **Check Persistence Configuration**: Ensure that your application is configured to use a persistent store (like a database) for workflow definitions and instances. In-memory storage will not retain state across application restarts.

2. **Verify Database State**: If you are using a database, check the tables related to workflow definitions to ensure that the published state is being set correctly when you publish a workflow.

3. **Review Application Logs**: Look for any errors or warnings in your application logs that might indicate a problem with workflow state persistence.

4. **Update Elsa Packages**: If you are using an older version of Elsa, consider updating to the latest version, as there may have been bug fixes related to state persistence.

5. **Consult Elsa Documentation**: Review the Elsa documentation for guidance on configuring persistence providers and troubleshooting common issues.

6. **Check for Known Issues**: Look for any known issues on the Elsa GitHub repository or community forums that might be related to your problem.

7. **Reach Out for Support**: If you are unable to resolve the issue, consider reaching out to the Elsa community or support channels for assistance.

Remember to back up your workflow definitions and any related data before making changes to your application's configuration or updating packages.

If you need further assistance, please provide more details about your persistence configuration and any relevant logs or error messages you are encountering.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).

**Reply by Serg046** [reply]
`IsPublished` db field value is 1 when I stop the app and then it is 0 after I start the app back.

**Comment by sfmskywalker** [comment]
Which version are you using? I believe this may have been fixed, but would be good to double-check if you are already using the latest preview packages.

**Reply by Serg046** [reply]
Latest stable ones:
```
<PackageReference Include="Elsa" Version="3.1.3" />
<PackageReference Include="Elsa.CSharp" Version="3.1.3" />
<PackageReference Include="Elsa.EntityFrameworkCore" Version="3.1.3" />
<PackageReference Include="Elsa.EntityFrameworkCore.Sqlite" Version="3.1.3" />
<PackageReference Include="Elsa.Identity" Version="3.1.3" />
<PackageReference Include="Elsa.Scheduling" Version="3.1.3" />
<PackageReference Include="Elsa.WorkflowProviders.BlobStorage" Version="3.1.3" />
<PackageReference Include="Elsa.Workflows.Api" Version="3.1.3" />
```

**Reply by sfmskywalker** [reply]
Perhaps you can try the latest 3.2.0-preview.1688 to see if the issue exists there as well?
